### PR TITLE
Chores/awsdb

### PR DESCRIPTION
**Pull request by cgarretty**
# Problem
While the current app is suitable for local development, we needed to make it available to be deployed on AWS Elastic-beanstalk running a single docker container.
 
# Solution
We run 3 commands created as a post app deployment hook ([H/T Edward Samuel](https://stackoverflow.com/questions/31978961/django-migrations-with-docker-on-aws-elastic-beanstalk?noredirect=1&lq=1)) which allows us to run the commands to the docker container after deployment - instead of before. These three commands:

1. Migrate the database
2. Create a superuser
3. Collect static file to be served

## Change summary

- Added ```.ebextensions/df-migrate.config``` to handle bash scripts to the docker container
- Added production environment variables for database connection and turned debugging off in the production environment for security reasons.
- Added ```fec/management/commands/createsu.py``` to create superusers programmatically and without input.

## Steps to Verify
1. Add environment admin user password to production environment
2. Add 
```
User.objects.create_superuser("<USER-NAME>", "<EMAIL>",
                                          os.environ['<ENVIRONMENT-VALUE>'])
```
to ```fec/management/commands/createsu.py```
3. ```eb deploy``` to environment 
4. Go to http://confero-env.cvqp3jgbbb.us-east-2.elasticbeanstalk.com/admin
5. Add Politicians and Campaigns
6. See Changes on homepage.

## Screenshot (Optional)
image

# For Discussion
I couldn't figure out how to serve django static files, but that looks like it is [another ticket](https://stackoverflow.com/questions/7241688/django-admin-css-missing) [entirely](https://www.caktusgroup.com/blog/2014/11/10/Using-Amazon-S3-to-store-your-Django-sites-static-and-media-files/).

### CODE REVIEW

File: confero/settings.py [hunk 1]
```diff
@@ -76,16 +79,28 @@
...
-        'NAME': os.getenv("DB_NAME", "confero"),
-        'USER': os.getenv("DB_USER", "postgres"),
-        'PASSWORD': os.getenv("DB_PASSWORD", "postgres"),
-        'HOST': os.getenv("DB_HOST", "localhost"),
-        'PORT': os.getenv("DB_PORT", "5432"),
+if 'RDS_DB_NAME' in os.environ:
+    DATABASES = {
+        'default': {
+            'ENGINE': 'django.db.backends.postgresql',
+            'NAME': os.environ['RDS_DB_NAME'],
```
Comment (will-wow) [inline confero/settings.py]: It'd be nice to not have to duplicate all these config lines. What do you think about something like:

```python
db_prefix = 'RDS_DB_'  if 'RDS_DB_NAME' in os.environ else 'DB_'

'default': {
            'ENGINE': 'django.db.backends.postgresql',
            'NAME': os.environ[db_prefix + 'NAME'],
            ...
```

File: fec/management/commands/createsu.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from django.core.management.base import BaseCommand
+from django.contrib.auth.models import User
+import os
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        if not User.objects.filter(username="admin").exists():
+            User.objects.create_superuser("cgarretty", "admin@admin.com",
```
Comment (will-wow) [inline fec/management/commands/createsu.py]: I assume the first argument is the username. Should that just be `admin` or something?

File: .ebextensions/db-migrate.config [hunk 3]
```diff
@@ -0,0 +1,19 @@
+files:
+  "/opt/elasticbeanstalk/hooks/appdeploy/post/10_post_migrate.sh":
```
Comment (will-wow) [inline .ebextensions/db-migrate.config]: what's the `10_` refer to? Just trying to track what's going on

File: fec/management/commands/createsu.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+                                          os.environ['SUPERUSER_KEY'])
```
Comment (will-wow) [inline fec/management/commands/createsu.py]: Nice use of the environment!

Could you add `SUPERUSER_KEY=password` or something to the `.env.example` file, to remind us this exists? I like having every possible environment variable listed in there, it helps when debugging a production issue to know what variables _should_ exist

File: .ebextensions/db-migrate.config (hunk 3 above)
Comment (cgarretty) [reply .ebextensions/db-migrate.config]: AWS uses this dir to run other scripts. The scripts run in sort order post deployment. We prefixed the file with ```10_``` so it doesn't interfere with those other scripts.

File: confero/settings.py (hunk 1 above)
Comment (cgarretty) [reply confero/settings.py]: smart. I like it a lot.

**Comment by cgarretty** [comment]
Should all be good, now. Some slight confusion with CircleCI not having the right DB variables, but that's fixed now.

File: confero/settings.py [hunk 5]
```diff
@@ -75,15 +78,15 @@
...
+db_prefix = 'RDS_DB_' if 'RDS_DB_NAME' in os.environ else 'DB_'
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.postgresql',
-        'NAME': os.getenv("DB_NAME", "confero"),
-        'USER': os.getenv("DB_USER", "postgres"),
-        'PASSWORD': os.getenv("DB_PASSWORD", "postgres"),
-        'HOST': os.getenv("DB_HOST", "localhost"),
-        'PORT': os.getenv("DB_PORT", "5432"),
+        'NAME': os.environ[db_prefix + 'NAME'],
```
Comment (will-wow) [inline confero/settings.py]: Nice! How's you get circle working, though? `os.getenv` lets you pass a default value as a second parameter, which let me set the default `DB_NAME` to `confero`, which matched what Circle wanted. I'd think that you'd either have to keep those defaults with `os.environ(db_prefix + 'NAME', "confero")`, or else pass in something like this [here](https://github.com/project-confero/confero/blob/master/.circleci/config.yml#L6):

```bash
      - image: circleci/python:3.7
        environment:
          DB_NAME: confero
      - image: circleci/postgres
        environment:
          POSTGRES_DB: confero
          POSTGRES_USER: postgres
```

Or better yet, both, to make the circle setup more explicit. But it's obviously working already, so don't let this block you from merging if you want to!

**Pull request merged**